### PR TITLE
Changed rhel version for LSO must-gather container image from 8 to 9

### PR DESCRIPTION
**Pull request by reivaj84**
Version(s):
4.15+

Issue:

Link to docs preview:
https://79756--ocpdocs-pr.netlify.app/openshift-dedicated/latest/support/gathering-cluster-data.html
https://79756--ocpdocs-pr.netlify.app/openshift-enterprise/latest/support/gathering-cluster-data.html
https://79756--ocpdocs-pr.netlify.app/openshift-rosa/latest/support/gathering-cluster-data.html

QE review:
- [x] QE has approved this change.

Additional information:
Local Storage Operator must-gather image, since 4.15 , it is based on rhel9 instead of rhel8 which was for OCP 4.14 and lower versions.

### CODE REVIEW

**Review by dfitzmau — APPROVED** [review]
/remove-label peer-review-in-progress

/label peer-review-done

**Comment by dfitzmau** [comment]
Would be worthwhile getting QE approval to verify that RHEL 9 compatability exists for version 4.15 and later.

**Comment by reivaj84** [comment]
I have validated that the LSO must-gather container image based on rhel8 is not avaialble for the 4.15 version, while the rhel9 based one works and collects the requested information related to LSO.

Hope it helps


```
> oc get clusterversion; oc get csv -n openshift-local-storage
NAME      VERSION   AVAILABLE   PROGRESSING   SINCE   STATUS
version   4.15.20   True        False         14d     Cluster version is 4.15.20


NAME                                          DISPLAY         VERSION               REPLACES   PHASE
local-storage-operator.v4.15.0-202407120536   Local Storage   4.15.0-202407120536              Succeeded


> oc adm must-gather --image=registry.redhat.io/openshift4/ose-local-storage-mustgather-rhel8:v4.15
[must-gather      ] OUT Using must-gather plug-in image: registry.redhat.io/openshift4/ose-local-storage-mustgather-rhel8:v4.15
...
...
Error running must-gather collection:
    gather did not start for pod must-gather-62khc: unable to pull image: ImagePullBackOff: Back-off pulling image "registry.redhat.io/openshift4/ose-local-storage-mustgather-rhel8:v4.15"

Falling back to `oc adm inspect clusteroperators.v1.config.openshift.io` to collect basic cluster information.



-- With rhel9 based image

> oc adm must-gather --image=registry.redhat.io/openshift4/ose-local-storage-mustgather-rhel9:v4.15
[must-gather      ] OUT Using must-gather plug-in image: registry.redhat.io/openshift4/ose-local-storage-mustgather-rhel9:v4.15
...
...
[must-gather      ] OUT pod for plug-in image registry.redhat.io/openshift4/ose-local-storage-mustgather-rhel9:v4.15 created
[must-gather-8nw8t] POD 2024-07-31T20:19:30.254044257Z Gathering data for ns/openshift-local-storage...
[must-gather-8nw8t] POD 2024-07-31T20:19:30.423252949Z Warning: apps.openshift.io/v1 DeploymentConfig is deprecated in v4.14+, unavailable in v4.10000+
[must-gather-8nw8t] POD 2024-07-31T20:19:30.577386635Z Wrote inspect data to must-gather/.
```

**Comment by dfitzmau** [comment]
Thanks, @reivaj84 . The output validates RHEL 9. I'll let the merger decide if a QE review is needed. 

/label merge-review-needed

**Review by Srivaralakshmi — COMMENTED** [review]
lgtm

**Comment by Srivaralakshmi** [comment]
@dfitzmau Please get the QE approval on this and then request for the merge again. Thank you!

**Comment by dfitzmau** [comment]
I contacted Hemant Kumar for an SME review and Chao Yang for a QE review.

**Pull request merged**
